app/grandchallenge/serving/views.py

```diff
@@ -137,25 +139,10 @@ def serve_component_interface_value(
     except (MultipleObjectsReturned, ComponentInterfaceValue.DoesNotExist):
         raise Http404("No ComponentInterfaceValue found.")
 
-    for perm, lookup in (
-        ("algorithms.view_job", "outputs"),
-        ("algorithms.view_job", "inputs"),
-        ("archives.view_archiveitem", "values"),
-        ("reader_studies.view_displayset", "values"),
-    ):
-        # Q | Q filters are very slow, this potentially does several db calls
-        # but each is quite performant. Could be optimised later.
-        if (
-            get_objects_for_user(
-                user=user,
-                perms=perm,
-            )
-            .filter(**{lookup: civ})
-            .exists()
-        ):
-            return protected_storage_redirect(
-                name=civ.file.name, creator=user, component_interface_value=civ
-            )
+    if civ in get_component_interface_values_for_user(user=user):
```

Comment:
This loads all of the objects into memory and does a filter client side, will be a lot for users with a lot of files. This will offload everything to the db:

```suggestion
    if get_component_interface_values_for_user(user=user).filter(pk=civ.pk).exists():
```

Reply:
Good point. Thanks!